acquia/blt
# Ability to truncate history on git remotes

Changes proposed:
---------
A number of projects using BLT and ACE are taking an extremely long time to deploy once the artefact has been built. 

In projects where large numbers of front end and composer dependencies are present, the number of GIT objects is very high in the ACE repository. Our analysis shows that a large portion of the deployment time is spent simply checking out the repository. 

We have experimented with truncating the history on branches in the ACE repo to remove large number of objects in the GIT database. 

This has led to large reductions in deploy time for a branch. In one example ACSF Drupal 8 build the average deploy time has reduced from 40 minutes to 25 when employing this approach.

-

We have performed testing on a number of PS EMEA projects and reviewed these changes here https://github.com/nedsbeds/blt/pull/1 but are now looking for feedback from the BLT team about this approach  . 

This will not suit every project so the functionality must be triggered through an argument or via the configuration deploy: truncate_history

-

Steps to verify the solution:
-----------
(How does someone verify that this does what you think does?)

1. Trigger by passing `--truncate-history` to the `blt artifact:deploy`
2. Verify the branch you have deployed only has a single commit on the ACE GIT repo
3. The Github repo will not have been altered.
